### PR TITLE
chore: Remove enzyme :bye:

### PR DESCRIPTION
**Pull request by koba04**
## Overview

This PR removes `enzyme` and migrates to ReactDOM, ReactTestRenderer, and `jest-styled-components`

## What I did

- Remove `enzyme` and related packages
- Use `jest-styled-components` to includes styles in snapshot files
  - If we don't need to include styles into snapshot files, I remove `jest-styled-components` from `devDependencies`.
  - The new format of a snapshot file does not include a snapshot of a context object but it seems not to be helpful.
- Rewrite unit tests using enzyme with ReactDOM and ReactTestRenderer
  - I wrote tests with ReactDOM as possible as I can because it is more robust than ReactTestRenderer that depends on React Component structures rather than DOM structures.

### CODE REVIEW

File: src/components/FlashMessage/FlashMessage.test.tsx [hunk 1]
```diff
@@ -1,27 +1,40 @@
-import { mount } from 'enzyme'
+// import { create } from 'react-test-renderer'
```
Comment (Tokky0425) [inline src/components/FlashMessage/FlashMessage.test.tsx:1]: This should be removed 👌

```suggestion
```

**Review by Tokky0425 — APPROVED** [review]
I left a comment about a very tiny correction,  but the rest is LGTM 🎉

File: src/components/FlashMessage/FlashMessage.test.tsx (hunk 1 above)
Comment (koba04) [reply src/components/FlashMessage/FlashMessage.test.tsx]: Thanks! 👍

**Review by Tokky0425 — DISMISSED** [review]
LGTM! 🌮

**Review by cidermitaina — APPROVED** [review]
LGTM🎉

**Pull request merged**